HW3/task2.py

```diff
@@ -0,0 +1,30 @@
+# В большой текстовой строке подсчитать количество встречаемых слов и вернуть 10 самых частых. 
+# Не учитывать знаки препинания и регистр символов. 
+# За основу возьмите любую статью из википедии или из документации к языку.
+
+string = "Высокоуровневый язык программирования общего назначения с динамической строгой типизацией \
+        и автоматическим управлением памятью, ориентированный на повышение производительности разработчика,\
+        читаемости кода и его качества, а также на обеспечение переносимости написанных на нём программ[27].\
+        Язык является полностью объектно-ориентированным в том плане, что всё является объектами[25]. \
+        Необычной особенностью языка является выделение блоков кода пробельными отступами[28]. Синтаксис ядра \
+        языка минималистичен, за счёт чего на практике редко возникает необходимость обращаться к \
+        документации[27]. Сам же язык известен как интерпретируемый и используется в том числе для \
+        написания скриптов[25]. Недостатками языка являются зачастую более низкая скорость работы и \
+        более высокое потребление памяти написанных на нём программ по сравнению с аналогичным кодом, \
+        написанным на компилируемых языках, таких как C или C++[25][27]."
+
+words_list = string.replace('.','').replace(',', '').replace('[', ' ').replace(']', ' ').lower().split()
```

Comment:
Используйте переменную `punctuation` из модуля `string`. Однако тогда `replace` будет использовать неудобно, потому что пунктуационных символов достаточно много. Нужно использовать цикл.
Куда проще будет использовать метод строк isalpha при переборе символов текста, пропуская кроме букв еще и пробел.

Reply:
Хорошо, попробую через цикл с использованием isalpha

Reply:
Примерно так?
```suggestion
string_list = string.split()
list_words = []
for word in string_list:
    if word.isalpha() or word == " ":
        list_w.append(word)
print(list_words)
```

Reply:
Или вообще не создавать список, а сразу в строку добавлять?
```suggestion
new_string = ''
for word in string:
    if word.isalpha() or word == " ":
        new_string += word
print(new_string)
```
Так лучше будет? А уже потом это строку сплитовать на список и создавать из него словарь